noxfile.py

```diff
@@ -170,3 +172,73 @@ def docs_serve(session: Session) -> None:
         shutil.rmtree(build_dir)
 
     session.run("sphinx-autobuild", *args)
+
+
+@nox.parametrize("replay_file_path", glob.glob("./e2e-tests/cookiecutters/*.json"))
+@session(python=main_python_version)
+def test_cookiecutter(session: Session, replay_file_path) -> None:
+    """Uses the tap template to build an empty cookiecutter.
+
+    Runs the lint task on the created test project.
+    """
+    args = session.posargs or ["1"]
+
+    cc_build_path = "/tmp"  # noqa: S108
+    folder_base_path = "./cookiecutter"
+
+    target_folder = (
+        "tap-template"
+        if Path(replay_file_path).name.startswith("tap")
+        else "target-template"
+    )
+    tap_template = Path(folder_base_path + "/" + target_folder).resolve()
+    replay_file = Path(replay_file_path).resolve()
+
+    if not Path(tap_template).exists():
+        return
+
+    if not Path(replay_file).is_file():
+        return
+
+    sdk_dir = Path(Path(tap_template).parent).parent
+    cc_output_dir = Path(replay_file_path).name.replace(".json", "")
+    cc_test_output = cc_build_path + "/" + cc_output_dir
+
+    if Path(cc_test_output).exists():
+        session.run("rm", "-fr", cc_test_output, external=True)
+
+    session.install(".")
+    session.install("cookiecutter", "pythonsed")
+
+    session.run(
+        "cookiecutter",
+        "--replay-file",
+        str(replay_file),
+        str(tap_template),
+        "-o",
+        cc_build_path,
+    )
+    os.chdir(cc_test_output)
```

Comment:
[`Session.chdir`](https://nox.thea.codes/en/stable/config.html#nox.sessions.Session.chdir) might be safer

Reply:
Thanks! Already fixed.